src/csvcubed/writers/qbwriter.py

```diff
@@ -284,18 +311,123 @@ def _generate_foreign_keys_for_cube(self) -> List[dict]:
 
         return foreign_keys
 
-    def _generate_virtual_columns_for_cube(self) -> List[Dict[str, Any]]:
+    def _generate_virtual_columns_for_obs_val_in_pivoted_shape_cube(
+        self,
+    ) -> List[Dict[str, Any]]:
+        virtual_columns: List[dict] = []
+
+        # Generate the virtual col defining the `?sliceUri rdf:type qb:Slice` triple.
```

Comment:
There's one thing that I did mention that seems to have been forgotten. Can you make sure to add some suitable debug logging statements at appropriate locations to help with diagnosing end-user bugs? 

The main place of concern for this is qbwriter.py.